pallets/staking/src/subject_metadata.rs

```diff
@@ -0,0 +1,1098 @@
+                let before = collator_state.total_counted;
+                let in_top = collator_state.increase_stake::<T>(
+                    &candidate_id,
+                    staker_id.clone(),
+                    before_amount,
+                    balance_amt,
+                )?;
+                let after = collator_state.total_counted;
+                if collator_state.is_active() && (before != after) {
+                    Pallet::<T>::update_active(candidate_id.clone(), after);
+                }
+                <CandidateInfo<T>>::insert(&candidate_id, collator_state);
+                let new_total_staked = <Total<T>>::get().saturating_add(balance_amt);
+                <Total<T>>::put(new_total_staked);
+                let nom_st: StakerMetadata<T::AccountId, BalanceOf<T>> = self.clone().into();
+                <StakerInfo<T>>::insert(&staker_id, nom_st);
+                Pallet::<T>::deposit_event(Event::StakeIncreased {
+                    staker: staker_id,
+                    candidate: candidate_id,
+                    amount: balance_amt,
+                    in_top,
+                });
+                return Ok(())
+            }
+        }
+        Err(Error::<T>::NoSuchStake.into())
+    }
+
+    /// Updates the bond locks for this staker.
+    ///
+    /// This will take the current self.total and ensure that a lock of the same amount is applied
+    /// and when increasing the bond lock will also ensure that the account has enough free balance.
+    ///
+    /// `additional_required_balance` should reflect the change to the amount that should be locked if
+    /// positive, 0 otherwise (e.g. `min(0, change_in_total_bond)`). This is necessary because it is
+    /// not possible to query the amount that is locked for a given lock id.
+    pub fn adjust_bond_lock<T: Config>(
+        &mut self,
+        additional_required_balance: StakeAdjust<Balance>,
+    ) -> DispatchResult
+    where
+        BalanceOf<T>: From<Balance>,
+        T::AccountId: From<AccountId>,
+    {
+        match additional_required_balance {
+            StakeAdjust::Increase(amount) => {
+                ensure!(
+                    <Pallet<T>>::get_staker_stakable_free_balance(&self.id.clone().into())
+                        >= amount.into(),
+                    Error::<T>::InsufficientBalance,
+                );
+
+                // additional sanity check: shouldn't ever want to lock more than total
+                if amount > self.total {
+                    log::warn!("LOGIC ERROR: request to reserve more than bond total");
+                    return Err(DispatchError::Other("Invalid additional_required_balance"))
+                }
+            },
+            StakeAdjust::Decrease => (), // do nothing on decrease
+        };
+
+        if self.total.is_zero() {
+            T::Currency::remove_lock(STAKER_LOCK_ID, &self.id.clone().into());
+        } else {
+            T::Currency::set_lock(
```

Comment:
Should rather be transferring the assets in for executors and marking the lock in pallet staking.